spec/command/apply_template_spec.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Command::ApplyTemplate do
+  context "when any template does not exist" do
+    let!(:app) { dummy_test_app }
+
+    it "raises error", :fast do
+      result = run_cpl_command("apply-template", "gvc", "rails", "unexistent", "-a", app)
+
+      expect(result[:status]).to eq(1)
+      expect(result[:stderr]).to include("Missing templates")
+      expect(result[:stderr]).to include("- unexistent")
+    end
+  end
+
+  context "when all templates exist" do
+    let!(:app) { dummy_test_app }
+
+    after do
+      run_cpl_command!("delete", "-a", app, "--yes")
+    end
+
+    it "applies valid templates", :fast do
+      result = run_cpl_command("apply-template", "gvc", "rails", "-a", app)
+
+      expect(result[:status]).to eq(0)
+      expect(result[:stderr]).to include("Created items")
+      expect(result[:stderr]).to include("- [app] #{app}")
+      expect(result[:stderr]).to include("- [identity] #{app}-identity")
+      expect(result[:stderr]).to include("- [workload] rails")
+    end
+
+    it "fails to apply invalid templates", :fast do
+      result = run_cpl_command("apply-template", "invalid", "-a", app)
+
+      expect(result[:status]).to eq(1)
+      expect(result[:stderr]).to include("Failed to apply templates")
+      expect(result[:stderr]).to include("- invalid")
+    end
+
+    it "applies valid templates and fails to apply invalid templates", :fast do
+      result = run_cpl_command("apply-template", "gvc", "invalid", "rails", "-a", app)
+
+      expect(result[:status]).to eq(1)
+      expect(result[:stderr]).to include("Created items")
```

Comment:
@dzirtusss why are we using stderr for progress and status messages? Why not use stdout?

Reply:
See https://www.perplexity.ai/search/what-are-the-p2EQeuORQPyYDSLAjJ0RXQ

> Use stderr for error messages, warnings, diagnostic information, and other output that is not the primary data being output by the command. This allows the actual data from stdout to be cleanly separated from the error/info messages.

Reply:
Given Sergey's response in Slack, I think we should keep it like this. Or if you want me to change it, we should do it in a separate PR.

Reply:
please keep as is here. even if we will decide to change (which I doubt, because this is default behaviour of many shell tools) better do it in another PR as this one is huge already